packages/fetchai/connections/http_client/connection.py

```diff
@@ -0,0 +1,276 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""HTTP client connection and channel"""
+
+import asyncio
+import logging
+from asyncio import CancelledError
+from typing import Optional, Set, cast
+
+import requests
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.mail.base import Address, Envelope
+
+from packages.fetchai.protocols.http.message import HttpMessage
+from packages.fetchai.protocols.http.serialization import HttpSerializer
+
+SUCCESS = 200
+NOT_FOUND = 404
+REQUEST_TIMEOUT = 408
+SERVER_ERROR = 500
+
+logger = logging.getLogger(__name__)
+
+RequestId = str
+
+
+class HTTPClientChannel:
+    """A wrapper for a HTTPClient for a RESTful API."""
+
+    def __init__(
+        self,
+        agent_address: Address,
+        address: str,
+        port: int,
+        excluded_protocols: Optional[Set[PublicId]] = None,
+        restricted_to_protocols: Optional[Set[PublicId]] = None,
+    ):
+        """
+        Initialize a channel and process the initial API specification from the file path (if given).
+
+        :param agent_address: the address of the agent.
+        :param address: RESTful API hostname / IP address
+        :param port: RESTful API port number
+        :param excluded_protocols: this connection cannot handle messages adhering to any of the protocols in this set
+        :param restricted_to_protocols: this connection can only handle messages adhering to protocols in this set
+        """
+        self.agent_address = agent_address
+        self.address = address
+        self.port = port
+        self.restricted_to_protocols = restricted_to_protocols
+        self.in_queue = None  # type: Optional[asyncio.Queue]  # pragma: no cover
+        self.loop = None  # type: Optional[asyncio.AbstractEventLoop]  # pragma: no cover
+        self.excluded_protocols = excluded_protocols
+        self.is_stopped = True
+        logger.info("Initialised the HTTP client channel")
+
+    def connect(self):
+        """Connect."""
+        if self.is_stopped:
+            self.is_stopped = False
```

Comment:
`self.is_stopped = False` needs to be moved after the assert on line 86

Reply:
Done